apps/frontend/src/pages/CreateCharity.tsx

```diff
@@ -0,0 +1,651 @@
+  const addOrganization = () => {
+    setNewCharityObject((prevState) => {
+      const updatedOrganizations = [...prevState.organizations];
+      const newOrganization: OrganizationObject = {
+        name: "",
+        logoUrl: "",
+        donated: 0,
+        receiving: false,
+      };
+      updatedOrganizations.push(newOrganization);
+
+      return {
+        ...prevState,
+        organizations: updatedOrganizations,
+      };
+    });
+  };
+
+  // Uploads the images to memory
+  const handleImageUpload = async (
+    index: number,
+    event: ChangeEvent<HTMLInputElement>,
+  ) => {
+    if (event.target.files?.[0]) {
+      const imageString = await imageBlobToBase64(event.target.files?.[0]);
+      const newImage: ImageObject = {
+        index: index,
+        image: event.target.files?.[0],
+        imageString: imageString,
+      };
+      setPartnerImages((prevState) => [...prevState, newImage]);
+    }
+  };
+
+  // Get the org image from the given index
+  const getImageFromOrgIndex = (index: number) => {
+    for (let i = 0; i < partnerImages.length; i++) {
+      if (partnerImages[i].index == index) {
+        return partnerImages[i].imageString;
+      }
+    }
+    return undefined;
+  };
+
+  // Converts the image file blob to base64 for preview
+  const imageBlobToBase64 = (
+    image: File | null,
+  ): Promise<string | undefined> => {
+    return new Promise((resolve, reject) => {
+      if (image) {
+        const reader = new FileReader();
+        reader.onload = () => {
+          resolve(reader.result as string);
+        };
+        reader.onerror = (error) => {
+          reject(error);
+        };
+        reader.readAsDataURL(image);
+      } else {
+        resolve(undefined);
+      }
+    });
+  };
+
+  return (
+    <>
+      <SearchBar />
+      <Container>
+        <Card sx={{ marginTop: "32px", padding: "10px" }}>
+          <CardContent>
+            <Typography variant={"h5"}>Create Charity</Typography>
+            <Box>
+              <form noValidate autoComplete="off" onSubmit={handleSubmit}>
+                <FormControl
+                  sx={{ width: "100%", padding: "10px", gap: "10px" }}
+                >
+                  <TextField
+                    type="text"
+                    label="Charity Title"
+                    id="charity-title"
+                    onChange={updateCharityName}
+                    value={newCharityObject.name}
+                  />
+                  <TextField
+                    type="text"
+                    multiline
+                    label="Charity Description"
+                    id="charity-description"
+                    onChange={updateDescription}
+                    value={newCharityObject.description}
+                  />
+                  <img
+                    src={logoImageString}
+                    style={{
+                      display: logoImageString ? "block" : "none",
```

Comment:
We should probably be conditionally displaying this element based on logoImageString rather than setting it to have a css display property of "none". Less html to load and in theory a user could remove the none property and then see the image element (second reason is probably not that relevant). Same pattern is used a few times

Reply:
Tried this, but for whatever reason it doesn't actually render the image, it just renders the base64 string instead so I think I'll leave it alone so I don't spend too much time on it